README.md

```diff
@@ -749,6 +784,14 @@ log-targets:
     # service name with a minus (e.g. '-svc1') to remove a previously added
     # service. '-all' will remove all services.
     services: [<service names>]
+
+    # (Optional) A list of key/value pairs defining labels which should be set
+    # on the outgoing logs. These labels are added to (protocol-dependent)
+    # default labels (e.g. 'pebble_service' for Loki).
```

Comment:
`pebble_service` is Loki-specific.

Syslog already has specialised fields which are more appropriate for attaching the service name (either [APP-NAME, PROCID, or MSGID](https://www.rfc-editor.org/rfc/rfc5424#page-14)).

Loki has no such mechanism. Hence, we **need** to attach a custom label so that Loki can know *where* the logs came from. This was the original rationale for the `pebble_service` label.

Reply:
Understood, but there are two further concerns here: the first is that although we're talking about ˋpebble_service`, the implication of what you're saying is no labels at all would be sent because it's unsupported, and the second is that there is actually a standard syslog way to send key/value pairs. From that [same spec](https://www.rfc-editor.org/rfc/rfc5424.html#page-15):

> STRUCTURED-DATA provides a mechanism to express information in a well defined, easily parseable and interpretable data format.  There are multiple usage scenarios.  For example, it may express meta-information about the syslog message or application-specific information such as traffic counters or IP addresses.
>
> (...)
> 
>    All examples in this section show only the structured data part of the message.  Examples should be considered to be on one line.  They are wrapped on multiple lines in this document for readability purposes.  A description is given after each example.
>
> Example 1 - Valid
> 
> [exampleSDID@32473 iut="3" eventSource="Application" eventID="1011"]

So in other words, there's a bit of inconsistency in the plan. Namingly:

1. Are labels supported at all for syslog?
2. If yes, then why would we not send `pebble_service` as well?
3. If not, then `pebble_service` can still be global, it's just that `syslog` doesn't support labels.

Reply:
Yes, the SD (structured data) fields is the standard way to send key/value pairs to syslog. However, as Jordan mentioned, there's also the APP-NAME field which is a more specific field for the "application name", which seems appropriate for the service name (not PROCID or MSGID). I think making this target type-specific (so Loki-specific in this instance) is the right thing to do here, as other log target types we add in future may have different needs.

Normally you can filter syslog logs by the various fields like APP-NAME. *However*, I think it's fine to research syslog further later when we actually add support for the syslog type. If we end up deciding not to use APP-NAME for some reason and instead use syslog structured data, that's okay.

For this specific documentation change in the plan description, let's leave out the second sentence altogether here ("These labels are added...") and instead add type-specific details under the "type" field, something like so:

```
    # (Required) The type of log target, which determines the format in
    # which logs will be sent. The supported types are:
    #
    # - loki: Use the Grafana Loki protocol. A "pebble_service" label is
    #         added automatically, with the name of the Pebble service as its value.
    type: loki
```

Later, we'd add a new entry `- syslog` with details about the APP-NAME field.

Reply:
@benhoyt I just updated the README as you described.